### DISCUSSION TITLE
What is velox's memory management mechanism?

### DISCUSSION THREAD

**Discussion by jiangjiguang** [facebookincubator/velox#4893] [Q&A]
I am very confused on veloxs memory management mechanism,
If I have given 25G memory by node.memory_gb=25 in node.properties, velox how to use theses memory.
Do velox keep theses memory as static ? it means put the 25G memory in a pool, and the program has to get memory from the pool ?

**Comment by whutjs** [comment]
I  am not familiar with the detail implementation of memory management in Velox, but I think many memory management mechanisms are basically the same. You can refer to [jemalloc](https://www.facebook.com/notes/10158791475077200/) for how memory management mechanism works.

**Comment by kgpai** [comment]
cc : @xiaoxmeng
